workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -385,12 +385,61 @@ def _update_es_node(es_node):
             source['main_file_name'] = ''
 
 
+def _assert_clean_elasticsearch(es):
+    exception = NonRecoverableError('Manager is not clean')
```

Comment:
Yes - with a better name for the method. and perhaps write a more informative message that it is only allowed to restore a snapshot on a clean manager.